.github/workflows/pre_publish.yml

```diff
@@ -65,9 +71,62 @@ jobs:
         with:
           tool: git-cliff
 
+      # TODO does this actually work?
       - uses: chainguard-dev/actions/setup-gitsign@0cda751b114eb55c388e88f7479292668165602a # v1.0.2
 
-      - id: pr-create
+      - name: Bump Kotlin version
+        if: inputs.package == 'iota-kotlin-sdk'
+        run: |
+          source ./.github/bump_version.sh
+
+          # Read current version from build.gradle.kts
+          CONFIG="bindings/kotlin/build.gradle.kts"
+          CURRENT_VERSION=$(grep '^version = ' $CONFIG | sed 's/version = "\(.*\)"/\1/')
+          echo "Current version: $CURRENT_VERSION"
+
+          # Bump the version using the input level
+          NEW_VERSION=$(bump_semver "$CURRENT_VERSION" "${{ inputs.level }}")
+          echo "New version: $NEW_VERSION"
+
+          # Update the version in build.gradle.kts
+          sed -i "s/version = \"$CURRENT_VERSION\"/version = \"$NEW_VERSION\"/" "$CONFIG"
+          echo "Updated build.gradle.kts with new version: $NEW_VERSION"
+
+          # Generate changelog for Kotlin
+          ./.github/gen-changelog.sh bindings/kotlin "$NEW_VERSION"
+
+          # Set output for the next step
+          echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+
+      - name: Bump Python version
+        if: inputs.package == 'iota-python-sdk'
+        run: |
+          source ./.github/bump_version.sh
+
+          # Read current version from pyproject.toml
+          CONFIG="bindings/python/src/pyproject.toml"
+          CURRENT_VERSION_PEP440=$(grep '^version = ' $CONFIG | sed 's/version = "\(.*\)"/\1/')
+          CURRENT_VERSION=$(pep440_to_semver "$CURRENT_VERSION_PEP440")
+          echo "Current version: $CURRENT_VERSION"
+
+          # Bump the version using the input level
+          NEW_VERSION=$(bump_semver "$CURRENT_VERSION" "${{ inputs.level }}")
+          echo "New version: $NEW_VERSION"
+
+          # Update the version in pyproject.toml
+          NEW_VERSION_PEP440=$(semver_to_pep440 "$NEW_VERSION")
+          sed -i "s/version = \"$CURRENT_VERSION_PEP440\"/version = \"$NEW_VERSION_PEP440\"/" "$CONFIG"
+          echo "Updated pyproject.toml with new version: $NEW_VERSION_PEP440"
+
+          # Generate changelog for Python
+          ./.github/gen-changelog.sh bindings/python "$NEW_VERSION"
+
+          # Set output for the next step
+          echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+
+      - name: Create PR for Rust
+        if: startsWith(inputs.package, 'iota-sdk')
```

Comment:
maybe move this to an ENV var? then we have this logic in one place at least

Reply:
Think I prefer to keep it that way